minio/credentials/providers.py

```diff
@@ -187,7 +187,11 @@ def retrieve(self):
 class IAMProvider(Provider):
     """IAM EC2 credential provider."""
 
-    def __init__(self, endpoint=None, http_client=None):
+    def __init__(self,
+                 endpoint=None,
+                 http_client=None,
+                 expiry_delta=timedelta(seconds=0)):
```

Comment:
given that a `timedelta` is mutable, it should not be used as a default value here. it would be better to set this to `None` and do `if expiry_delta is None: expiry_delta = ...`

Reply:
expiry_delta should be 10secs default and can be changed to a lower of higher value as necessary.

Reply:
I'm just referring to https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments, not the value itself

Reply:
Gotcha, I changed the default argument value to be `None`. If that is the case we then set `expiry_time` to 10 secs